dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/HibernateTrackedEntityStore.java

```diff
@@ -588,57 +593,81 @@ private String getFromSubQueryJoinProgramOwnerConditions(TrackedEntityQueryParam
+   * <p>If a program is specified, the join is based on program ownership (PO). If no program is
+   * specified, the join is based either on program ownership or the tracked entity's registering
+   * unit.
+   *
+   * <p>The specific JOIN conditions depend on the {@code ouMode}:
+   *
+   * <ul>
+   *   <li>{@code DESCENDANTS} – matches organisation units using the org unit's PATH
+   *   <li>{@code CHILDREN} – matches the org unit's PATH or any of its immediate children
+   *   <li>{@code SELECTED} – matches the specified org unit ID directly
+   *   <li>{@code ALL} – no org unit constraints are applied
+   * </ul>
+   *
+   * <p>If the org unit mode is not {@code ALL} and the user has no superuser privileges, the method
+   * also ensures that the tracked entity owner falls within the appropriate access scope (either
+   * search or capture).
+   *
+   * @return a SQL INNER JOIN clause for organisation units
    */
-  private String getFromSubQueryJoinOrgUnitConditions(TrackedEntityQueryParams params) {
+  private String getFromSubQueryJoinOrgUnitConditions(
+      TrackedEntityQueryParams params, MapSqlParameterSource sqlParams) {
     StringBuilder orgUnits = new StringBuilder();
 
-    handleOrganisationUnits(params);
+    UserDetails userDetails = getCurrentUserDetails();
+    Set<OrganisationUnit> effectiveSearchOrgUnits =
+        getOrgUnitsFromUids(userDetails.getUserEffectiveSearchOrgUnitIds());
+    Set<OrganisationUnit> captureScopeOrgUnits =
+        getOrgUnitsFromUids(userDetails.getUserOrgUnitIds());
+
+    if (params.isOrganisationUnitMode(ACCESSIBLE)) {
+      params.setOrgUnits(effectiveSearchOrgUnits);
+      params.setOrgUnitMode(DESCENDANTS);
+    } else if (params.isOrganisationUnitMode(CAPTURE)) {
+      params.setOrgUnits(captureScopeOrgUnits);
+      params.setOrgUnitMode(DESCENDANTS);
+    }
 
     orgUnits
         .append(" INNER JOIN organisationunit OU ")
         .append("ON OU.organisationunitid = ")
         .append(getOwnerOrgUnit(params));
 
-    if (!params.hasOrganisationUnits()) {
-      return orgUnits.toString();
+    if (params.hasOrganisationUnits()) {
+      if (params.isOrganisationUnitMode(DESCENDANTS)) {
+        orgUnits.append(getDescendantsQuery(params));
+      } else if (params.isOrganisationUnitMode(CHILDREN)) {
+        orgUnits.append(getChildrenQuery(params));
+      } else if (params.isOrganisationUnitMode(SELECTED)) {
+        orgUnits.append(getSelectedQuery(params));
+      }
     }
 
-    if (params.isOrganisationUnitMode(OrganisationUnitSelectionMode.DESCENDANTS)) {
-      orgUnits.append(getDescendantsQuery(params));
-    } else if (params.isOrganisationUnitMode(OrganisationUnitSelectionMode.CHILDREN)) {
-      orgUnits.append(getChildrenQuery(params));
-    } else if (params.isOrganisationUnitMode(OrganisationUnitSelectionMode.SELECTED)) {
-      orgUnits.append(getSelectedQuery(params));
+    if (params.isOrganisationUnitMode(ALL) || getCurrentUserDetails().isSuper()) {
```

Comment:
We had this validation in the old code to check if the user has super privileges, which I believe is incorrect.
As far as I know, we only skip sharing validations when the user is a superuser, but we never skip ownership validations, unless the org unit mode is explicitly set to `ALL`, so I should remove it from here.
@ameenhere?

Reply:
There's a bit of confusion when it comes to this, since an admin user is not a super user.
A superuser is the one that has the authority `ALL`, and that's usually the system user, the admin user has other privileges, but not this one. 
That means, in our current logic, we would skip the ownership validations for the system user, but not for admin users.

Is that correct?
If it is, then we are saying, since you are a superuser, we assume you have capture scope access to all org units in the system, so we skip the ownership validations. Same goes for sharing validations. Is that it?